### ISSUE TITLE
YeastMine: GOAnnotation.evidence is empty

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#2090]
YeastMine does not have GOAnnotation.evidence, it's not populated but it's in the model:

https://github.com/yeastgenome/yeastmine-bio-sources/blob/master/yeastmine-global_additions.xml

Tell @kkarra

**Comment by julie-sullivan** [comment]
`evidence` was moved to `OntologyAnnotation` (instead of `GOAnnotation`) in a recent intermine release:

```
<class name="OntologyAnnotation" is-interface="true" term="http://semanticscience.org/resource/SIO_001166">
	<attribute name="qualifier" type="java.lang.String" term="http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#C41009"/>
	<reference name="subject" referenced-type="Annotatable" reverse-reference="ontologyAnnotations"/>
	<reference name="ontologyTerm" referenced-type="OntologyTerm" reverse-reference="ontologyAnnotations"/>
	<collection name="evidence" referenced-type="OntologyEvidence"/>
	<collection name="dataSets" referenced-type="DataSet"/>
</class>
```

Try removing `GOAnnotation.evidence` from your data model?

**Comment by julie-sullivan** [comment]
@kkarra if I download and load the yeast file from here: http://current.geneontology.org/products/pages/downloads.html
will that replicate your problem?

**Comment by kkarra** [comment]
Sure, files from there should be good to use.(I do load directly from the database, but, it doesn't matter for the issue here.)

I removed evidence from GOAnnotation, but, that didn't seem to help. So, then I checked core.xml and noticed that evidence was missing in yeastmine-global-additions.xml version of OntologyAnnotation. So, I added it there and went through the steps again but didn't seem to help either.

Am not able to locate where core.xml and global-additions.xml are mentioned to merge (or not??) with the y-g-a.xml file.  

I am using 4.0.0 - and it looks like flymine and humanmine are also on 4.0.0 (had to use intermine registry to figure out the version). If I look in 
https://github.com/intermine/flymine-bio-sources/blob/master/flymine-global_additions.xml
GOAnnotation does have evidence.. So, I am not sure how these model changes play out..
https://intermineorg.wordpress.com/2017/09/08/intermine-2-0-proposed-model-changes-iii/

it does look like these were in there since 2.0 ? 

I feel positive about seeing it work as expected couple week ago. I switched to using 4.0.0 instead of 4.0.+ to avoid unknown changes being applied to the code/builds. Is that the correct way to go about that.

**Comment by julie-sullivan** [comment]
![yeast-go](https://user-images.githubusercontent.com/1663057/65021187-87b0b400-d926-11e9-842d-7742a72962c3.png)

Using flymine, I loaded that data file ONLY, and evidence was populated correctly.

This means that the yeastmine data model is likely the issue.

**Comment by julie-sullivan** [comment]
The [GO annotation source](https://github.com/intermine/intermine/blob/dev/bio/sources/go-annotation/src/main/resources/go-annotation_additions.xml) has these:

1. `GOEvidenceCode` extends `OntologyAnnotationEvidenceCode`
2. `GOEvidence` extends `OntologyEvidence`

Therefore if you use the go annotation source, those will be in your data model. But those are NOT in the data model for `yeastmine.dev`. 

Are you using a different go annotation source?

**Comment by julie-sullivan** [comment]
>  I switched to using 4.0.0 instead of 4.0.+ to avoid unknown changes being applied to the code/builds. Is that the correct way to go about that.

Yes that's fine! However, update to use 4.0.1 because I fixed a bug.

**Comment by julie-sullivan** [comment]
Never mind, I found it. [go-annotation-db](https://github.com/yeastgenome/yeastmine-bio-sources/blob/master/go-annotation-db/src/main/resources/go-annotation-db_additions.xml)

You are saying this used to work? and populate the evidence collection correctly?

**Comment by kkarra** [comment]
Yes, I believe I saw it working once when I thought keys were the problem and thought I had fixed it. But, next time around it was gone.

I tried to fix the model, latest is here..

https://github.com/yeastgenome/yeastmine-bio-sources/blob/master/yeastmine-global_additions.xml

**Comment by kkarra** [comment]
which property file has core, genomic additions ,so inclusion?

**Comment by julie-sullivan** [comment]
@kkarra can you remove this too:

https://github.com/yeastgenome/yeastmine-bio-sources/blob/master/yeastmine-global_additions.xml#L38

ontologyAnnotations is on Annotatable, not BioEntity now.

**Comment by julie-sullivan** [comment]
@kkarra GOEvidence and GOEvidenceCode are still incorrect.

https://github.com/yeastgenome/yeastmine-bio-sources/blob/master/yeastmine-global_additions.xml#L202

**Comment by julie-sullivan** [comment]
> which property file has core, genomic additions ,so inclusion?

What do you mean? Here is where those files are configured:

https://github.com/yeastgenome/yeastmine/blob/master/dbmodel/build.gradle#L38

The `mergeModel` task merges them all together. 

is that what you mean?

**Comment by kkarra** [comment]
Yes, model was wrong. adding extends OntologEvidence and OntologyEvidenceCode helped. I see evidence collection. Still need to run a whole build and check it out. thanks!
